table/block_based/block_based_table_reader.cc

```diff
@@ -1531,14 +1534,26 @@ BlockBasedTable::MaybeReadBlockAndLoadToCache(
           TBlocklike::kBlockType != BlockType::kFilter &&
           TBlocklike::kBlockType != BlockType::kCompressionDictionary &&
           rep_->blocks_maybe_compressed;
+      // This flag, if true, tells BlockFetcher to return the uncompressed
+      // block when ReadBlockContents() is called.
       const bool do_uncompress = maybe_compressed;
       CompressionType contents_comp_type;
       // Maybe serialized or uncompressed
       BlockContents tmp_contents;
+      BlockContents uncomp_contents;
+      BlockContents comp_contents;
```

Comment:
Are you worried about the cost of default construction? I did this to simplify the code, since ```BlockFetcher::ReadBlockContents()``` can return the compressed or uncompressed block depending on various conditions, so seemed simpler to just track them separately.

Reply:
> Are you worried about the cost of default construction?

Yes, I was concerned about three buckets for two items, but I guess each one is just 24 bytes, so probably not a significant cost. I could see it being tricky to simplify further, so ok to put that off.

I'll try to run a quick test to measure any regression on the standard single-cache configuration "hot path".